oak_remote_attestation_verification/src/verifier.rs

```diff
@@ -70,25 +79,161 @@ impl AttestationVerifier for InsecureAttestationVerifier {
+        let default_binary_verifier = opts
+            .default_transparency_verification_options
+            .as_ref()
+            .map(create_transparency_verifier);
+
+        let mut binary_verifiers = BTreeMap::new();
+        opts.transparency_verification_options
+            .iter()
+            .for_each(|(name, opt)| {
+                binary_verifiers.insert(name.to_owned(), create_transparency_verifier(opt));
+            });
+        // TODO(#3641): Set other verifiers, e.g., reference value verifiers.
+        let verifiers = Vec::new();
+        Self {
+            verifiers,
+            binary_verifiers,
+            default_binary_verifier,
+        }
+    }
+}
+
+impl AttestationVerifier for AttestationLayerVerifier {
+    fn verify(
+        &self,
+        evidence: &AttestationEvidence,
+        endorsement: &AttestationEndorsement,
+        reference_value: &ReferenceValue,
+    ) -> anyhow::Result<()> {
+        for verifier in &self.verifiers {
+            verifier.verify(evidence, endorsement, reference_value)?;
+        }
+        // TODO(#3641): Update accordingly once #4074 is completed.
+        for verifiers in self.binary_verifiers.values() {
+            for verifier in verifiers {
+                verifier.verify(evidence, endorsement, reference_value)?;
+            }
+        }
+        //  TODO(#3641): Run the following for binaries not specified in `binary_verifiers`.
+        if let Some(verifiers) = &self.default_binary_verifier {
+            for verifier in verifiers {
+                verifier.verify(evidence, endorsement, reference_value)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Creates a list of AttestationVerifiers based on the given verification options.
+fn create_transparency_verifier(
+    opts: &TransparencyVerificationOptions,
+) -> Vec<Box<dyn AttestationVerifier>> {
+    let mut verifiers: Vec<Box<dyn AttestationVerifier>> = Vec::new();
+    verifiers.push(Box::new(EndorsementStatementVerifier {}));
+    if let Some(VerificationData(data)) = &opts.rekor_entry_verification {
+        verifiers.push(Box::new(RekorLogEntryVerifier {
+            base64_pem_encoded_rekor_public_key: data.base64_pem_encoded_rekor_public_key.clone(),
+            base64_pem_encoded_endorser_public_key: data
+                .base64_pem_encoded_endorser_public_key
+                .clone(),
+        }))
+    }
+    verifiers
+}
+
+/// AttestationVerifier for verifying the content of an endorsement statement and checking that it
+/// has the same binary digest as the measurement.
+pub struct EndorsementStatementVerifier {
+    // TODO(#3641): Additional info may be required here to correctly extract the binary digest from
+    // the given evidence.
+}
+
+impl AttestationVerifier for EndorsementStatementVerifier {
+    fn verify(
+        &self,
+        evidence: &AttestationEvidence,
+        endorsement: &AttestationEndorsement,
+        _reference_value: &ReferenceValue,
```

Comment:
I don't think endorsements should be verified against reference values, i.e. this parameter shouldn't be here at all.